pkg/server/controller.go

```diff
@@ -129,6 +207,65 @@ func (ctrl *Controller) drainMiddleware(next http.HandlerFunc) http.HandlerFunc
 	}
 }
 
+func (ctrl *Controller) authMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		jwtCookie, err := r.Cookie(jwtCookieName)
+		if err != nil {
+			ctrl.httpServer.ErrorLog.Printf("There seems to be problem with jwt token cookie: %v", err)
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		if jwtCookie == nil {
+			ctrl.httpServer.ErrorLog.Printf("Missing jwt cookie")
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
```

Comment:
that can be done, but I'm wondering is this even the error handling we want? Do we maybe want to show a different `Unauthorized` page to user instead of `/login`?

Reply:
A very good question.

I'd rather add a flash message saying "not authorised" for such a case. There is one more point we should always keep in mind dealing with auth: user should not know the exact reason of the denial – in most of cases such a message with detailed explanation of what went wrong implies disclosure.